### PR TITLE
Modified the outbound flow for guest calling

### PR DESCRIPTION
**Pull request by adhmenon**
# COMPLETES #< [SPARK-52966](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-529662)>

## This pull request addresses

* Guest calling support for Calling SDK.

## by making the following changes

* Currently in the callind SDK, there was no support for guest calling outbound flows.
* Code has been added so that it checks the outbound flows for a guest user (ie in that case destination address will be empty), and based on that removes the callee object from the post request load.
* Unit tests have also been added.

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

1. Make Call Flow (Posting request to mobius)
2. End Call Flow (Delete request to mobius)
3. Media Request 
All the above was done to properly test the entire call flow.

* Open the sample app via the amplify link given - [Sample App](https://pr-3794.d3m3l2kee0btzx.amplifyapp.com/)
* Inside the sample app, click on 'Calling' in the samples
* Select 'guest user' in credentials - 
Screenshot 2024-08-26 at 4 45 00 PM
* Click on 'Generate guest token' and you should see a guest token generated.
Screenshot 2024-08-26 at 4 47 13 PM
* Go this this site [JWT Site](https://jwt.io/) and copy the jwe token (from the encoded area). Paste it into the sample app's JWE token for destination.
* In the sample, app press on 'Initialize calling'
* Once that is done, press on 'register calling'
* Now open up the network logs.
* Once it has been registered, click on the following button  (it should auto enable without any address)- 
Screenshot 2024-08-26 at 4 49 40 PM
* You should now see in the network logs a post request for mobius gets sent with the following payload (similar type0 - 
Screenshot 2024-08-26 at 4 53 25 PM
* Now, proceed to end the call. In the network logs, you should see a DELETE request that gets sent to the mobius URI.

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [x] I discussed changes with code owners prior to submitting this pull request

- [x] I have not skipped any automated checks
- [x] All existing and new tests passed
- [x] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: docs/samples/calling/app.js
Comment (adhmenon) [inline docs/samples/calling/app.js:191]: There are changes from Shreyas's PR here as well. Will highlight my changes only.

File: packages/calling/src/CallingClient/calling/call.ts
Comment (adhmenon) [inline packages/calling/src/CallingClient/calling/call.ts:199]: Made this an optional so that guest calling does not send the object.

File: packages/calling/src/CallingClient/calling/call.ts
Comment (adhmenon) [inline packages/calling/src/CallingClient/calling/call.ts:2095]: Only if the destination is empty, we remove `callee` object payload (as per requirement).

File: packages/calling/src/CallingClient/line/index.ts
Comment (adhmenon) [inline packages/calling/src/CallingClient/line/index.ts:72]: Had to add this private variable over here so that we can access the `GUEST_CALLING` enum value. This is so that we can check for it in the `makeCall()` flow.

File: packages/calling/src/CallingClient/line/index.ts
Comment (adhmenon) [inline packages/calling/src/CallingClient/line/index.ts:271]: Added the 'if' case over here, in case the destination is not present it moves here and then sets up guest calling flow.
Nit: Will need to change the log to include 'guest calling call' probably. Will make the change.

File: packages/calling/src/CallingClient/line/line.test.ts
Comment (adhmenon) [inline packages/calling/src/CallingClient/line/line.test.ts:339]: Just added some tests to test the latest changes.

File: packages/calling/src/CallingClient/line/index.ts
Comment (Kesari3008) [inline packages/calling/src/CallingClient/line/index.ts:109]: One way to avoid creating a new private variable is to move the serviceData declaration under the constructor outside and make it a global variable with a default value of Calling. And within the constructor it will get updated with the value received in the parameter

**Review by Kesari3008 — CHANGES_REQUESTED** [review]
Please upload the logs once testing is finished

File: packages/calling/src/CallingClient/calling/callManager.ts
Comment (adhmenon) [reply packages/calling/src/CallingClient/calling/callManager.ts:175]: Yup, this is the reason.

**Comment by adhmenon** [comment]
Have temporarily commented out some tests so that we get the PR deployed on Amplify. Will revert these once we have successfully tested it.

File: packages/calling/src/CallingClient/calling/call.ts
Comment (Kesari3008) [inline packages/calling/src/CallingClient/calling/call.ts:2075]: We should have some sort of UT even to test different payload being sent out of this post request

**Review by Kesari3008 — APPROVED** [review]
Approving with 1 minor comment. Please add the UT request before merge

**Review by Shreyas281299 — APPROVED** [review]

**Pull request merged**
